### DISCUSSION TITLE
feat(extras): markdown -> change markdown previewer

### DISCUSSION THREAD

**Discussion by andreacfromtheapp** [LazyVim/LazyVim#4257] [Extra Requests]
Hi,

[markdown-preview](https://github.com/iamcco/markdown-preview.nvim) is unmaintained (last release 2022) and stopped working (with Node 20). Besides, it has (too!) many dependencies, some unmaintained too. Besides not being able to use it, this sounds like a supply chain attack or disaster waiting to happen.

## Possible alternatives

- [glow](https://github.com/ellisonleao/glow.nvim) is nice but I would not switch to it, because markdown-preview works in a browser.

- I found a potential candidate, that does both "in nvim" and browser,  in [peek](https://github.com/toppair/peek.nvim?tab=readme-ov-file)

Cheers

ps: I could test and PR if necessary.

**Comment by dpetka2001** [comment]
`peek` was considered as default Markdown previewer in the past (see this PR #1718), but consequentially  was not taken into account because of an old issue that required an old version of `deno` (at the time I built it with 1.25.4 specifically) so that it could be built correctly. That issue has since been fixed and now that I just tested it with a new LazyVim default installation, it correctly builds with latest `deno`.

However, I did find [this issue](https://github.com/toppair/peek.nvim/issues/65) that seems to have problems with `deno` when installed via a linux distro package manager (not sure if it affects all linux distros or just some). In my case I installed `deno` via [mise](https://github.com/jdx/mise) and it does not have the problem mentioned in the issue. I'm mentioning this, just in case you decide to go forward with a PR, so you can test it thoroughly.

**Reply by andreacfromtheapp** [reply]
thanks :) I wasn't aware of any of this.

**Reply by andreacfromtheapp** [reply]
I tried it but I hate having yet another "server" installed. The browser view works only on chrome, which I don't use. going with glow instead. Since I already use it with other tools anyway.

If anyone else wants to give it a go.

**Reply by dpetka2001** [reply]
Strange I use Firefox and when I set `app = "browser"`, it correctly opens the Markdown file in Firefox.

**Reply by andreacfromtheapp** [reply]
point taken. I use Safari on macOS and I didn't bother trying. Perhaps I'll give it another go. Right now the idea of installing deno gives me shivers. Not cause of deno. I want to reduce the amount of deps I need.

**Reply by dpetka2001** [reply]
Yeah I can understand that. Just stick to whatever suits your needs better. `glow` it is for you :smile:

**Reply by andreacfromtheapp** [reply]
So, I tried peek again and it works fine. Could only "test" with macOS and deno installed with homebrew and it works. I have no Linux nor Windows machines to make sure. All in all it still seems a good candidate.

**Reply by dpetka2001** [reply]
I've tested `peek.nvim` on Linux Mint 21.3 and it works, not sure about Windows. Just out of curiosity what exactly is not working for you with `markdown-preview`. I don't mean to discourage you from making a PR (maintainer has the final say after all), but in my case it works correctly with Node 20.17.0.

**Reply by andreacfromtheapp** [reply]
It won't build anymore. iirc it stopped working for me at some point, with noice just showing a node version pop up.

<img width="1085" alt="Screenshot 2024-08-26 at 11 19 49 AM" src="https://github.com/user-attachments/assets/6a19ddfc-9a5b-40a5-a6d4-0178e6efdadb">

**Reply by dpetka2001** [reply]
There's #4196 which will fix that. Just waiting to be merged. In the meantime you could incorporate that change in the `markdown-preview` spec in your personal configuration and it will build correctly. Just change the `build` key to be the same as in the PR.

**Reply by dpetka2001** [reply]
Like I said in my previous post, this is not meant to discourage you from submitting a PR about `peek.nvim`. The maintainer will decide in the end. In my opinion, `peek.nvim` is a good candidate for replacing `markdown-preview`, but I'm not making any decisions.

**Reply by andreacfromtheapp** [reply]
given that markdown-preview last release was in 2022, and that judging by stale issues and PRs it seems abandoned, peek would be a nice candidate indeed. Let's see what folke thinks about this when he is back.

**Reply by dpetka2001** [reply]
If you submitted a PR, it would grab Folke's attention more I believe. After he comes back from vacation, I'm sure he'll have a lot to deal with all his plugins (especially `noice.nvim` seems to have some kind of regression), so he might just brush off discussion threads etc and forget about it.

**Reply by andreacfromtheapp** [reply]
fair point. I'll open one in the next few days. I still begrudge the extra deno dep tho. We'll see what he thinks.

Cheers for now! thanks for your help :)
